pkg/controller/addon/addon_controller.go

```diff
@@ -0,0 +1,361 @@
+package addon
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	mf "github.com/jcrossley3/manifestival"
+	"github.com/prometheus/common/log"
+	op "github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	pConfig "github.com/tektoncd/operator/pkg/controller/config"
+	"io/ioutil"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"golang.org/x/xerrors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"path/filepath"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	"time"
+)
+
+var (
+	ctrlLog = logf.Log.WithName("ctrl").WithName("addon")
+	ErrPipelineNotReady = xerrors.Errorf("tekton-pipelines not ready")
```

Comment:
Why are we exporting this?

Reply:
we are not using it outside this package, at present. So im making it unexported now. In the future if we want to use the error in tests we will have to export it.